File: src/window-info.js
var windowInfo = require("../build/Release/window-info");

const title = "League of Legends (TM) Client";

const list = windowInfo.listWindows();
console.log(list.map(windowInfo.windowTitle));

setInterval(() => {
  console.log(`${title} is fullscreen? `, windowInfo.isFullScreen(title));
}, 2000);

File: src/window-info.cc
#include "window-info.h"
#include <napi.h>

Napi::Object Init(Napi::Env env, Napi::Object exports)
{
  exports.Set(Napi::String::New(env, "listWindows"),
              Napi::Function::New(env, ListWindows));
  exports.Set(Napi::String::New(env, "windowTitle"),
              Napi::Function::New(env, WindowTitle));
  return exports;
}

NODE_API_MODULE(windowInfo, Init)